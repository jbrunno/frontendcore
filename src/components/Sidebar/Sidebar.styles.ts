import { css, styled } from '@mui/material/styles';
import Drawer from '../Drawer/Drawer';

type SidebarDrawerStyledProps = {
  isExpanded: boolean;
  sidebarWidth: number;
  expandedSidebarWidth: number;
};

export const SidebarDrawerStyled = styled(Drawer, {
  shouldForwardProp: (prop) => !['isExpanded'].includes(prop.toString()),
})<SidebarDrawerStyledProps>(
  ({ theme, isExpanded, sidebarWidth, expandedSidebarWidth }) => css`
    width: ${theme.fn.size(isExpanded ? expandedSidebarWidth : sidebarWidth)};

    .MuiDrawer-paper {
      width: ${theme.fn.size(isExpanded ? expandedSidebarWidth : sidebarWidth)};
      padding: ${theme.fn.spacing(16, 8)};
      transition: width 0.3s;
      background-color: ${theme.palette.common.white};
      display: grid;
      grid-template-rows: auto 1fr auto;
      row-gap: ${theme.fn.spacing(16)};
      overflow: hidden;
      border: 0;
      box-shadow: ${isExpanded
        ? '0 3px 15px rgba(0, 0, 0, 0.08), 0 3px 10px rgba(0, 0, 0, 0.08)'
        : '0 3px 15px rgba(0, 0, 0, 0.02), 0 3px 10px rgba(0, 0, 0, 0.02)'};
    }
  `,
);
