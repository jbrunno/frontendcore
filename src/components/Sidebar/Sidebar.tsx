import { ReactNode } from 'react';
import { SidebarUser } from '../SidebarUser/SidebarUser';
import { DrawerProps } from '../Drawer/Drawer';
import { SidebarLogo } from '../SidebarLogo/SidebarLogo';
import { useSidebar } from './hooks/useSidebar';
import { SidebarDrawerStyled } from './Sidebar.styles';

type ChildrenProps = {
  isExpanded: boolean;
  handleExpanded: (expanded: boolean) => void;
};

type SidebarProps = {
  children?: ((props: ChildrenProps) => ReactNode) | ReactNode;
} & Omit<DrawerProps, 'children'>;

export const SIDEBAR_WIDTH = 72;
export const EXPANDED_SIDEBAR_WIDTH = 240;

export function Sidebar({ children, ...props }: SidebarProps) {
  const { isExpanded, loggedUser, logout, handleExpanded } = useSidebar();
  return (
    <SidebarDrawerStyled
      variant="permanent"
      anchor="left"
      onMouseEnter={() => handleExpanded(true)}
      onMouseLeave={() => handleExpanded(false)}
      sidebarWidth={SIDEBAR_WIDTH}
      expandedSidebarWidth={EXPANDED_SIDEBAR_WIDTH}
      isExpanded={isExpanded}
      {...props}
    >
      <SidebarLogo isExpanded={isExpanded} />
      {typeof children === 'function'
        ? children({ isExpanded, handleExpanded })
        : children}
      <SidebarUser
        isExpanded={isExpanded}
        handleLogOut={() => logout()}
        user={{ name: loggedUser }}
      />
    </SidebarDrawerStyled>
  );
}

export default Sidebar;
