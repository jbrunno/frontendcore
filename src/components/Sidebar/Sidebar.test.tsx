import { render } from '@test-utils/component-testing';
import React from 'react';
import { Sidebar } from './Sidebar';

test('renders the Sidebar', () => {
  const { container } = render(<Sidebar />);
  expect(container).toMatchSnapshot();
});
