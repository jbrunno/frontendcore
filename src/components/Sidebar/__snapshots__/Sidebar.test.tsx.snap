// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Sidebar 1`] = `
.emotion-1 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: calc(3.15rem + 21.599999999999998px);
}

.emotion-1 .MuiDrawer-paper {
  width: calc(3.15rem + 21.599999999999998px);
  padding: 1rem 0.5rem;
  -webkit-transition: width 0.3s;
  transition: width 0.3s;
  background-color: #fff;
  display: grid;
  grid-template-rows: auto 1fr auto;
  row-gap: 1rem;
  overflow: hidden;
  border: 0;
  box-shadow: 0 3px 15px rgba(0, 0, 0, 0.02),0 3px 10px rgba(0, 0, 0, 0.02);
}

.emotion-2 {
  background-color: #fff;
  color: #3B3B41DE;
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  box-shadow: none;
  overflow-y: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  z-index: 1200;
  -webkit-overflow-scrolling: touch;
  position: fixed;
  top: 0;
  outline: 0;
  left: 0;
  border-right: 1px solid #0000001F;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  height: calc(2.275rem + 15.6px);
  margin: 1rem 0rem;
  padding: 0rem 0.375rem;
}

.emotion-5 {
  max-height: 100%;
  height: calc(1.8375rem + 12.6px);
}

.emotion-5:nth-of-type(2) {
  margin-left: 0.375rem;
  height: calc(2.275rem + 15.6px);
}

.emotion-7 {
  display: grid;
  grid-template-columns: auto;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-column-gap: 0.5rem;
  column-gap: 0.5rem;
  padding: 0rem 0.5rem;
}

.emotion-10 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 40px;
  height: 40px;
  font-family: Lato,sans-serif;
  font-size: 1.25rem;
  line-height: 1;
  border-radius: 50%;
  overflow: hidden;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: #fafafa;
  background-color: #bdbdbd;
  grid-row: span 2;
}

<div>
  <div
    class="MuiDrawer-root MuiDrawer-docked emotion-0 emotion-1"
    expandedsidebarwidth="240"
    sidebarwidth="72"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiDrawer-paper MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft emotion-2"
    >
      <div
        class="emotion-3 emotion-4"
      >
        <img
          alt="Logo CRM"
          class="emotion-5 emotion-6"
          src="logo.svg"
        />
      </div>
      <div
        class="emotion-7 emotion-8"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault emotion-9 emotion-10"
        >
          NN
        </div>
      </div>
    </div>
  </div>
</div>
`;
